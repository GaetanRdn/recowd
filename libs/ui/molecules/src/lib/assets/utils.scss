@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
