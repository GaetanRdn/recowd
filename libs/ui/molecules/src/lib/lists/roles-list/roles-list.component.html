<rc-list>
  <rc-list-item *ngFor="let role of roles; trackBy: trackByFn">
    <rc-line>{{ role.titre }}</rc-line>
    <rc-line>{{ role.contractType }}</rc-line>
    <rc-line>{{ role | fromToDate }}</rc-line>
  </rc-list-item>
</rc-list>
