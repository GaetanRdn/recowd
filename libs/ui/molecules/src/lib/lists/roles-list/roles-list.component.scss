@use '../../assets/utils';

@mixin rc-roles-list($config) {
  $main-content: utils.map-get-deep($config, typography, body-bold);
  $secondary-content: utils.map-get-deep($config, typography, body-regular);
  $primary: utils.map-get-deep($config, colors, primary);

  rc-roles-list {
    .rc-line:nth-child(1) {
      @include utils.use-typography($main-content);
    }

    .rc-line:not(:nth-child(1)) {
      color: darken(map-get($primary, color), 30%);

      @each $key in font-size, font-family, font-weight, line-height {
        #{$key}: map-get($secondary-content, $key);
      }
    }
  }
}
