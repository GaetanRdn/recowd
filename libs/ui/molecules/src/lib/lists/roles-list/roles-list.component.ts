import {
  ChangeDetectionStrategy,
  Component,
  Input,
  TrackByFunction,
} from '@angular/core';
import { CommonModule } from '@angular/common';
import { Role } from '@recowd/models';
import { ListModule } from '@recowd/ui-atoms';
import { FromToDatePipeModule } from '@recowd/pipes';

@Component({
  selector: 'rc-roles-list',
  standalone: true,
  imports: [CommonModule, ListModule, FromToDatePipeModule],
  templateUrl: './roles-list.component.html',
  styleUrls: ['./roles-list.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RolesListComponent {
  @Input() public roles: Role[] = [];

  protected trackByFn: TrackByFunction<Role> = (_: number, role: Role) =>
    role.titre + role.startDate;
}
