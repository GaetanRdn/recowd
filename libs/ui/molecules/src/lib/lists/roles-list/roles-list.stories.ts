import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { RolesListComponent } from './roles-list.component';

export default {
  title: 'molecules/lists/roles-list',
  decorators: [
    moduleMetadata({
      imports: [RolesListComponent],
    }),
  ],
} as Meta<RolesListComponent>;

const Template: Story<RolesListComponent> = (args: RolesListComponent) => ({
  props: {
    roles: args.roles,
  },
  template: `<rc-roles-list [roles]="roles"></rc-roles-list>`,
});

export const Default = Template.bind({});
Default.args = {
  roles: [
    {
      titre: 'Lead',
      startDate: '2020-01-01',
      endDate: null,
      contractType: 'FREELANCE',
    },
    {
      titre: 'Dev',
      startDate: '2019-01-01',
      endDate: '2019-12-31',
      contractType: 'CDI',
    },
  ],
};
