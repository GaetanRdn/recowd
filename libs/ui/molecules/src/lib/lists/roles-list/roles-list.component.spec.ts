import { TestBed, waitForAsync } from '@angular/core/testing';
import { RolesListComponent } from './roles-list.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';
import {
  LEAD_DEV_CDI_ESN_ROLE,
  LEAD_DEV_FREELANCE_ROLE,
} from '@recowd/test/data';

describe('RolesListComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [RolesListComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create empty', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());

  it('should create with roles', () => {
    templateLookup.hostComponent.roles = [
      LEAD_DEV_FREELANCE_ROLE,
      LEAD_DEV_CDI_ESN_ROLE,
    ];

    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-roles-list [roles]="roles"></rc-roles-list>`,
})
class HostComponent {
  public roles: RolesListComponent['roles'] = [];
}
