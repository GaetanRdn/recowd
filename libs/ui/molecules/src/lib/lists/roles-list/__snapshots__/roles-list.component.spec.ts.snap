// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolesListComponent should create empty 1`] = `
<rc-roles-list>
  <rc-list
    role="list"
  >
    <span
      class="rc-list-items-wrapper"
    >
      
    </span>
  </rc-list>
</rc-roles-list>
`;

exports[`RolesListComponent should create with roles 1`] = `
<rc-roles-list>
  <rc-list
    role="list"
  >
    <span
      class="rc-list-items-wrapper"
    >
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        <rc-line
          class="rc-line"
        >
          Lead developer
        </rc-line>
        <rc-line
          class="rc-line"
        >
          FREELANCE
        </rc-line>
        <rc-line
          class="rc-line"
        >
          January 2020 - July 2022
        </rc-line>
      </rc-list-item>
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        <rc-line
          class="rc-line"
        >
          Lead developer
        </rc-line>
        <rc-line
          class="rc-line"
        >
          CDI ESN
        </rc-line>
        <rc-line
          class="rc-line"
        >
          January 2020 - December 2020
        </rc-line>
      </rc-list-item>
      
    </span>
  </rc-list>
</rc-roles-list>
`;
